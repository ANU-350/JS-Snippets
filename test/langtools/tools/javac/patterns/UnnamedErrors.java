/**
 * @test /nodynamiccopyright/
 * @bug 8304246
 * @summary Compiler Implementation for Unnamed patterns and variables
 * @enablePreview
 * @compile/fail/ref=UnnamedErrors.out -XDrawDiagnostics -XDshould-stop.at=FLOW UnnamedErrors.java
 */
public class UnnamedErrors {
    private int _; // error
    private int _, x;  // error
    private int x, _, y, _, z, _;  // error
    private int _ = 0, _ = 1; // error
    private int a = 0, _ = 1; // error

    record R(int _) {} //no record components
    UnnamedErrors(int _) {} //no constructor parameters
    void test(int _) {} //no method parameters

    record RR(int x) {}
    void test2() {
        Object o = Integer.valueOf(42);
        if (o instanceof _) {} //no top level

        if (o instanceof _(int x)) {} //no record pattern head

        switch (o) {
            case _:
                System.out.println("no underscore top level");
            default:
                System.out.println("");
        }

        switch (o) {
            case var _:
                System.out.println("no var _ top level");
            default:
                System.out.println("");
        }
    }

    void dominance_error(Object o) {
        switch (o) {
            case Number _ ->
                    System.out.println("A Number");
            case Integer _, String _ ->             // Error - dominated case pattern: `Integer _`
                    System.out.println("An Integer or a String");
            default ->
                    System.out.println("rest");
        }
    }

    void mixed_named_unnamed_error(Object o) {
        switch (o) {
            case Integer i, String _ ->
                    System.out.println("named/unnamed");
            default ->
                    System.out.println("rest");
        }

        switch (o) {
            case Integer _, String s ->
                    System.out.println("unnamed/named");
            default ->
                    System.out.println("rest");
        }

        switch (o) {
            case PairIS(_, _), String s ->
                    System.out.println("unnamed patterns/named");
            default ->
                    System.out.println("rest");
        }
    }

    private void test1() {
        try (Lock _ = null) {
        } catch (_) { }
    }

    int guard_errors(Object o, int x1, int x2) {
        return switch (o) {
            case Integer _ when x1 == 2, String _ when x2 == 1 -> 1;
            default -> 2;
        };
    }

    class Lock implements AutoCloseable {
        @Override
        public void close() {}
    }
    record PairIS(int i, String s) {}
}
