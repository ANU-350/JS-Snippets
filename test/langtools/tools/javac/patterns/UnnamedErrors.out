UnnamedErrors.java:9:17: compiler.err.underscore.as.identifier
UnnamedErrors.java:10:17: compiler.err.underscore.as.identifier
UnnamedErrors.java:11:20: compiler.err.underscore.as.identifier
UnnamedErrors.java:11:26: compiler.err.underscore.as.identifier
UnnamedErrors.java:11:32: compiler.err.underscore.as.identifier
UnnamedErrors.java:12:17: compiler.err.underscore.as.identifier
UnnamedErrors.java:12:24: compiler.err.underscore.as.identifier
UnnamedErrors.java:13:24: compiler.err.underscore.as.identifier
UnnamedErrors.java:15:18: compiler.err.underscore.as.identifier
UnnamedErrors.java:16:23: compiler.err.underscore.as.identifier
UnnamedErrors.java:17:19: compiler.err.underscore.as.identifier
UnnamedErrors.java:22:26: compiler.err.underscore.as.identifier
UnnamedErrors.java:24:26: compiler.err.underscore.as.identifier
UnnamedErrors.java:27:18: compiler.err.underscore.as.identifier
UnnamedErrors.java:34:18: compiler.err.restricted.type.not.allowed.here: var
UnnamedErrors.java:34:22: compiler.err.underscore.as.identifier
UnnamedErrors.java:77:18: compiler.err.underscore.as.identifier
UnnamedErrors.java:77:19: compiler.err.expected: token.identifier
UnnamedErrors.java:11:17: compiler.err.already.defined: kindname.variable, x, kindname.class, UnnamedErrors
UnnamedErrors.java:36:13: compiler.err.unconditional.pattern.and.default
UnnamedErrors.java:45:18: compiler.err.pattern.dominated
UnnamedErrors.java:54:29: compiler.err.flows.through.from.pattern
UnnamedErrors.java:61:29: compiler.err.flows.through.from.pattern
UnnamedErrors.java:68:32: compiler.err.flows.through.from.pattern
- compiler.note.preview.filename: UnnamedErrors.java, DEFAULT
- compiler.note.preview.recompile
24 errors